src/main/kotlin/ar/edu/unahur/obj2/vendedores/Vendedor.kt

```diff
@@ -2,50 +2,49 @@ package ar.edu.unahur.obj2.vendedores
 
 class Certificacion(val esDeProducto: Boolean, val puntaje: Int)
 
-abstract class Vendedor {
-  // Acá es obligatorio poner el tipo de la lista, porque como está vacía no lo puede inferir.
-  // Además, a una MutableList se le pueden agregar elementos
+abstract class Vendedor () {
   val certificaciones = mutableListOf<Certificacion>()
 
-  // Definimos el método abstracto.
-  // Como no vamos a implementarlo acá, es necesario explicitar qué devuelve.
   abstract fun puedeTrabajarEn(ciudad: Ciudad): Boolean
 
-  // En las funciones declaradas con = no es necesario explicitar el tipo
+  abstract fun esInfluyente() : Boolean
+
   fun esVersatil() =
     certificaciones.size >= 3
-      && this.certificacionesDeProducto() >= 1
-      && this.otrasCertificaciones() >= 1
-
-  // Si el tipo no está declarado y la función no devuelve nada, se asume Unit (es decir, vacío)
-  fun agregarCertificacion(certificacion: Certificacion) {
-    certificaciones.add(certificacion)
-  }
+    && this.certificacionesDeProducto() >= 1
+    && this.otrasCertificaciones() >= 1
 
   fun esFirme() = this.puntajeCertificaciones() >= 30
 
+  fun puntajeCertificaciones() = certificaciones.sumBy {it.puntaje}
+
   fun certificacionesDeProducto() = certificaciones.count { it.esDeProducto }
   fun otrasCertificaciones() = certificaciones.count { !it.esDeProducto }
+  fun esGenerico() = this.otrasCertificaciones() >= 1
 
-  fun puntajeCertificaciones() = certificaciones.sumBy { c -> c.puntaje }
-}
-
-// En los parámetros, es obligatorio poner el tipo
-class VendedorFijo(val ciudadOrigen: Ciudad) : Vendedor() {
-  override fun puedeTrabajarEn(ciudad: Ciudad): Boolean {
-    return ciudad == ciudadOrigen
+  fun agragarCertificacion (unaCertificacion: Certificacion) {
+    certificaciones.add(unaCertificacion)
   }
 }
 
-// A este tipo de List no se le pueden agregar elementos una vez definida
-class Viajante(val provinciasHabilitadas: List<Provincia>) : Vendedor() {
-  override fun puedeTrabajarEn(ciudad: Ciudad): Boolean {
-    return provinciasHabilitadas.contains(ciudad.provincia)
-  }
+class VendedorFijo (val ciudadDeOrigen: Ciudad) : Vendedor() {
+
+  override fun esInfluyente() = false
+
+  override fun puedeTrabajarEn(ciudad: Ciudad) = ciudad == ciudadDeOrigen
+
 }
 
-class ComercioCorresponsal(val ciudades: List<Ciudad>) : Vendedor() {
-  override fun puedeTrabajarEn(ciudad: Ciudad): Boolean {
-    return ciudades.contains(ciudad)
-  }
+class Viajante (val provinciasHabilitadas: List<Provincia>) : Vendedor() {
+
+  override fun esInfluyente() = provinciasHabilitadas.sumBy { it.poblacion } >= 10000000
+
+  override fun puedeTrabajarEn(ciudad: Ciudad) = provinciasHabilitadas.contains(ciudad.provincia)
 }
+
+class ComercioCorresponsal (val ciudades: List<Ciudad>) : Vendedor() {
+
+  override fun esInfluyente() = ciudades.size >= 5 || (ciudades.map { it.provincia }.toSet()).size >= 3
```

Comment:
No está mal, pero la parte que calcula las provincias podría estar en otro método, así te queda un cóodigo más legible, ejemplo: 

```suggestion
 fun provinciasEnLasQuePuedeTrabajar() = (ciudades.map { it.provincia }.toSet()).size
  override fun esInfluyente() = ciudades.size >= 5 || provinciasEnLasQuePuedeTrabajar() >= 3
```